### PR TITLE
Add support for host:0 in URIs (OS-chosen port)

### PR DESCRIPTION
**Pull request by jafyvilla**
zmq expects host:*

### CODE REVIEW

**Comment by eile** [comment]
But tcp://:0 works?

File: zeq/detail/broker.h [hunk 1]
```diff
@@ -14,11 +14,12 @@ std::string buildZmqURI( std::string host, const uint16_t port )
...
     if( host.empty( ))
         host = "*";
-    if( host != "*" && port == 0 ) // zmq does not support host:0
-        throw std::runtime_error(
-            "OS-chosen port not supported with hostname " + host );
 
-    return std::string( "tcp://" ) + host + ":" + std::to_string( int( port ));
+    const std::string tcpURI( "tcp://" + host );
+    if( host != "*" && port == 0 ) // zmq expects host:* instead of host:0
+        return tcpURI + ":*";
```
Comment (eile) [inline zeq/detail/broker.h]: Does tcp://:* also work? If yes, can be simplified to if( port == 0 )

Comment (jafyvilla) [inline zeq/detail/broker.h]: Mmmm... should the empty host (tcp://:*) work? I don't think it does, but tcp://*:* works fine as far as I can tell

**Comment by hernando** [comment]
AFAIR, tcp://:0 is not a valid URI and it will make Servus throw.

**Pull request merged**